pkg/composableschemadsl/lexer/flaggablelexer_test.go

```diff
@@ -0,0 +1,74 @@
+package lexer
+
+import (
+	"slices"
+	"testing"
+
+	"github.com/authzed/spicedb/pkg/composableschemadsl/input"
+)
+
+var flaggableLexerTests = []lexerTest{
+	{"use expiration", "use expiration", []Lexeme{
+		{TokenTypeKeyword, 0, "use", ""},
+		{TokenTypeWhitespace, 0, " ", ""},
+		{TokenTypeKeyword, 0, "expiration", ""},
+		tEOF,
+	}},
+	{"use expiration and", "use expiration and", []Lexeme{
+		{TokenTypeKeyword, 0, "use", ""},
+		{TokenTypeWhitespace, 0, " ", ""},
+		{TokenTypeKeyword, 0, "expiration", ""},
+		{TokenTypeWhitespace, 0, " ", ""},
+		{TokenTypeKeyword, 0, "and", ""},
+		tEOF,
+	}},
+	{"expiration as non-keyword", "foo expiration", []Lexeme{
+		{TokenTypeIdentifier, 0, "foo", ""},
+		{TokenTypeWhitespace, 0, " ", ""},
+		{TokenTypeKeyword, 0, "expiration", ""},
+		tEOF,
+	}},
+	{"and as non-keyword", "foo and", []Lexeme{
+		{TokenTypeIdentifier, 0, "foo", ""},
+		{TokenTypeWhitespace, 0, " ", ""},
+		{TokenTypeKeyword, 0, "and", ""},
+		tEOF,
+	}},
+	{"invalid use flag", "use foobar", []Lexeme{
+		{TokenTypeKeyword, 0, "use", ""},
+		{TokenTypeWhitespace, 0, " ", ""},
+		{TokenTypeIdentifier, 0, "foobar", ""},
+		tEOF,
+	}},
+	{"use flag after definition", "definition use expiration", []Lexeme{
+		{TokenTypeKeyword, 0, "definition", ""},
+		{TokenTypeWhitespace, 0, " ", ""},
+		{TokenTypeKeyword, 0, "use", ""},
+		{TokenTypeWhitespace, 0, " ", ""},
+		{TokenTypeKeyword, 0, "expiration", ""},
```

Comment:
Is it worth reworking these tests? `use` and `expiration` are now keywords, so the semantics of the tests don't completely scan.

Reply:
Yes, I think so; we should add them in as formal keywords in the lexer tests